### PR TITLE
Drivetrain

### PR DESCRIPTION
**Pull request by NinaPlazonic**
First attempt at creating a 2022 drivetrain for the protobot. 
Tested in-person by Karthik, Harris, Pranjal, Nina, and Kai. It works well, and we might need to change things later on.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        public static final double KI = 0.0d;
+        public static final double KD = 0.06d;
+        
+        public static final double CLOSED_LOOP_RAMP = 0.5;
+        public static final double MAX_VELOCITY = 21549;
+        
+        public static final double DEFAULT_MAX_VELOCITY_PERCENTAGE = 0.6;
+        public static final double DEFAULT_MAX_TURNING_SPEED = 0.5d;
+        public static final double VELOCITY_SLOWDOWN_MODIFIER = 0.25d;
+        public static final int LEFT_LEAD_TALON_CAN_ID = 0;
```
Comment (karthikappana35) [inline src/main/java/frc/robot/Constants.java:59]: Right Talon Lead should have CAN ID of 1 and follower motors don't exist anymore

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    private WPI_TalonFX leftTalonLead;
+    private WPI_TalonFX rightTalonLead;
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+    
+    /**
+    *
+    */
+    public Drivetrain() {
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: Lots of formatting issues here. Let's clean this up

**Review by bjmcternan — CHANGES_REQUESTED** [review]
Let's clean up the formatting and this will be good to me!

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    leftTalonLead = new WPI_TalonFX(2);
```
Comment (karthikappana35) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: Use constants and reverse left motor

**Review by karthikappana35 — CHANGES_REQUESTED** [review]
OOPS I forgot to push a commit!

**Review by KaiFrc — APPROVED** [review]
It works lol

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 2 above)
Comment (bjmcternan) [reply src/main/java/frc/robot/subsystems/Drivetrain.java]: Looks good!

**Review by bjmcternan — APPROVED** [review]
Looks good to me!

**Review by karthikappana35 — APPROVED** [review]
Updated Constant Usage and motor inversion

**Comment by srkolb** [comment]
Is someone trying to use CircleCI to build Team 293 code?  I received an approval request to allow a 3rd party application (CircleCI) to access Team 293.  I will approve it if it was deliberate.  Setting up a continuous integration pipeline is an admirable goal, though also very ambitious.  

Scott Kolb

**Pull request merged**